test/nn/test_parametrization.py

```diff
@@ -1414,6 +1414,26 @@ def fn(weight):
 
                         gradcheck(fn, (m.parametrizations.weight.original,))
 
+    def test_register_parametrization_no_grad(self):
+        r"""Test that it is possible to register a parametrization without gradient"""
+
+        class SplitAndCat(nn.Module):
+            def right_inverse(self, x):
+                # split the tensor in two halfs
+                return torch.split(x, x.shape[1] // 2)
+
+            def forward(self, x0, x1):
+                return torch.cat([x0, x1])
+
+        model = nn.Linear(8, 8)
+
+        model.weight.requires_grad = False
+        # One parametrization with unsafe=True
+        parametrize.register_parametrization(model, "weight", SplitAndCat())
+        # making sure the decomposed Tensors both have requires_grad == False
+        self.assertFalse(model.parametrizations.weight.original0.requires_grad)
+        self.assertFalse(model.parametrizations.weight.original1.requires_grad)
```

Comment:
Probably also check that
```suggestion
        self.assertFalse(model.weight.requires_grad)
```

Reply:
And check that this is respected when you assign to it?